sdk/eventgrid/Azure.Messaging.EventGrid/api/Azure.Messaging.EventGrid.netstandard2.0.cs

```diff
@@ -821,6 +1073,27 @@ public partial class IotHubDeviceTelemetryEventData : Azure.Messaging.EventGrid.
     {
         internal IotHubDeviceTelemetryEventData() { }
     }
+    [System.Runtime.InteropServices.StructLayoutAttribute(System.Runtime.InteropServices.LayoutKind.Sequential)]
+    public readonly partial struct JobStatus : System.IEquatable<Azure.Messaging.EventGrid.SystemEvents.JobStatus>
```

Comment:
I am worried that these names are a little too general.

What do you think about adding this transform:

```
### Some intermediate FarmBeats types have very general names, add a prefix

```yaml
directive:
  - from: swagger-document
    where: $.definitions.ResourceActionType
    transform: >
      $["x-ms-enum"]["name"] = "FarmBeatsResourceActionType";
  - from: swagger-document
    where: $.definitions.JobStatus
    transform: >
      $["x-ms-enum"]["name"] = "FarmBeatsJobStatus";
```

To add the "FarmBeats" prefix here?  I am thinking about doing this for TypeScript.  We could move this change upstream into the azure-rest-api-specs repository long term if we all agree on a name here (the FarmBeats prefix does seem pretty uniform here)

Reply:
Yeah good catch. I agree with the proposed names.